retrievalmarket/rtvllog/retrieval_log.go

```diff
@@ -58,6 +58,30 @@ func (r *RetrievalLog) OnQueryEvent(evt retrievalmarket.ProviderQueryEvent) {
 		"status", evt.Response.Status,
 		"msg", evt.Response.Message,
 		"err", evt.Error)
+
+	// Log failures to DB
+	st := &RetrievalDealState{
+		UnsealPrice: evt.Response.UnsealPrice,
+		Message:     evt.Response.Message,
+	}
+	if evt.Error != nil {
+		st.Status = evt.Error.Error()
+	} else {
```

Comment:
In the case that evt.Error is not nil, let's
- set st.Status to "Failed"
- check if evt.Response.Message is set
  - if not, set st.Message to evt.Error.Error()

Reply:
I think this will cause us to log all queries (even the successful ones as well). If evt.Error is not nil then response itself was successful but we might send "not available" or "error in query" responses along with "available" response. I would have no condition to decide when to log and when to skip. 
From the code it looks like evt.Response.Message is always set for query event and evt.Error.Error() would be empty for successful responses when "not available" or "error in query" message is being sent. Overriding the message value will result in los of the real reason for failure.